reform_cage.py

```diff
@@ -0,0 +1,176 @@
+	2) The topology of the cage.
+
+        3) Optional Arguments.
+        3.1) Optional (-c): Reform the unrelaxed cage from .mol file.
+        3.2) Optional (-b): Reform the unrelaxed building-blocks (bb).
+
+Usage: $ python reform_cage.py <file.mol>
+
+Author: James T. Pegg
+
+"""
+import argparse
+import stk
+from rdkit.Chem import AllChem
+import operator
+import os
+
+def reform_cage(name, result, deconstructed_topology):
+    """
+    Reform the unrelaxed two-component stk cage.
+
+    """
+    building_blocks = []
+    for k in result:
+      bb_mol = AllChem.MolFromSmiles(k)
+      bb_Hs = AllChem.AddHs(bb_mol)
+      AllChem.EmbedMolecule(bb_Hs)
+      if result[k][0] >= 3:
+          bb_su = stk.StructUnit3(bb_Hs, [result[k][1]])
+          if bb_su not in building_blocks:
+              building_blocks.append(bb_su)
+      else:
+          bb_fg = [result[k][1]]
+          bb_su = stk.StructUnit2(bb_Hs, [result[k][1]])
+          if bb_su not in building_blocks:
+              building_blocks.append(bb_su)
+
+    cage = stk.Cage(building_blocks, deconstructed_topology)
+    cage.write(name+'-out.mol')
+    cage.dump(name+'-out.json')
+    return cage
+
+
+def write_building_blocks(name, result):
+    """
+    Writes the optimised building blocks to .mol file.
+
+    """
+    os.mkdir('BuildingBlocks')
+    wbb_count = 1
+    for k in result:
+        bb_mol = AllChem.MolFromSmiles(k)
+        bb_Hs = AllChem.AddHs(bb_mol)
+        AllChem.EmbedMolecule(bb_Hs)
+        if result[k][0] >= 3:
+            bb_su = stk.StructUnit3(bb_Hs, [result[k][1]])
+            bb_su.write('BuildingBlocks/'+name+'-'+str(wbb_count)+'bb.mol')
+        else:
+            bb_su = stk.StructUnit2(bb_Hs, [result[k][1]])
+            bb_su.write('BuildingBlocks/'+name+'-'+str(wbb_count)+'bb.mol')
+        wbb_count += 1
+
+
+def topology_calc(coordination_numbers):
+    """
+    Finds the topology of the system.
+
+    """
+    if str(coordination_numbers) == '[(2, 3), (3, 2)]':
+        return stk.TwoPlusThree()
+    elif str(coordination_numbers) == '[(2, 6), (3, 4)]':
+        return stk.FourPlusSix()
+    elif str(coordination_numbers) == '[(2, 9), (3, 6)]':
+        return stk.SixPlusNine()
+    elif str(coordination_numbers) == '[(2, 12), (3, 8)]':
+        return stk.EightPlusTwelve()
+    elif str(coordination_numbers) == '[(3, 4), (4, 2)]':
+        return stk.TwoPlusFour()
+    elif str(coordination_numbers) == '[(3, 6), (4, 3)]':
+        return stk.ThreePlusSix()
+    elif str(coordination_numbers) == '[(3, 12), (4, 6)]':
+        return stk.SixPlusTwelve()
+    elif str(coordination_numbers) == '[(3, 8)]':
+        return stk.FourPlusFour()
+    elif str(coordination_numbers) == '[(3, 8), (4, 6)]':
+        return stk.FourPlusFour()
+    else:
+        return 'unknown'
+
+
+def resolve_functional_group(known_smiles, unknown_smiles, functional_group, unknown_functional_group, building_blocks_dict, smiles_fragments, uncorrected_smiles_fragments):
```

Comment:
This line is way too long. you can split your arguments so that there is one per line.